File: homeworks/Nigmatullin/sortings/insertionSort.cpp
#include <iostream>

using namespace std;

const int N = 100;

void insertionSort(int *arr, int n)
{
	for (int i = 1; i < n; i++)
	{
		int k = i - 1;
		int temp = arr[i];

		while (k >= 0 && arr[k] > temp)
		{
			arr[k + 1] = arr[k];
			arr[k] = temp;
			k--;
		}

	}
}

int main()
{
	int n, arr[N];
	cin >> n;

	for (int i = 0; i < n; i++)
	{
		cin >> arr[i];
	}

	insertionSort(arr, n);

	for (int i = 0; i < n; i++)
	{
		cout << arr[i] << " ";
	}

	return 0;
}

File: homeworks/Ivanova/II semestr/Sortings/Insertion_Sort.pas
Program Insertion_Sort;
uses crt;
const n = 1000;
type a = array[1..n] of integer;
var mas: a;
    i, temp, nom: integer;
    f: text;
    
Procedure InsertionSort(mas: a);
var i, j: integer;
begin
for i:=1 to n-1 do
                begin
                nom:=i+1;
                temp:=mas[nom];
for j:=i+1 downto 2 do
                    begin
if (temp<mas[j-1]) then
                   begin
                   mas[j]:=mas[j-1];
                   nom:=j-1;
                   end;
                    end;
                mas[nom]:=temp;
                end;
writeln('The sorted array: ');
for i:=1 to 1000 do 
                 write(mas[i], ' ');
end;

begin
assign(f,'ints.txt');
reset(f);
i:=0;
while not eof(f) do
                 begin
                 inc(i);
                 read(f,mas[i]);
                 end;
close(f);      
InsertionSort(mas);
readln;
end.